engine/src/main/java/org/terasology/rendering/dag/AbstractNode.java

```diff
@@ -15,43 +15,78 @@
  */
 package org.terasology.rendering.dag;
 
+import com.google.api.client.util.Maps;
 import com.google.common.collect.Sets;
+import java.util.Map;
 import java.util.Set;
+import org.terasology.assets.ResourceUrn;
 import org.terasology.registry.In;
+import org.terasology.rendering.opengl.BaseFBM;
+import org.terasology.rendering.opengl.DefaultDynamicFBOs;
 import org.terasology.rendering.opengl.FBOConfig;
-import org.terasology.rendering.opengl.FrameBuffersManager;
+import org.terasology.rendering.opengl.fbms.DynamicFBM;
+import org.terasology.rendering.opengl.fbms.StaticFBM;
 
 /**
  * TODO: Add javadocs
  */
 public abstract class AbstractNode implements Node {
+
+    @In
+    private StaticFBM staticFBM;
+
     @In
-    private FrameBuffersManager frameBuffersManager;
+    private DynamicFBM dynamicFBM;
 
     private Set<StateChange> desiredStateChanges;
-    private Set<String> fboNames;
+    private Map<ResourceUrn, Set<BaseFBM>> fboUsages;
 
     private NodeTask task;
     private RenderTaskListGenerator taskListGenerator;
 
     protected AbstractNode() {
         desiredStateChanges = Sets.newLinkedHashSet();
-        fboNames = Sets.newHashSet();
+        fboUsages = Maps.newHashMap();
     }
 
-    protected void requireFBO(FBOConfig fboConfig) {
+    protected void requireStaticFBO(FBOConfig fboConfig) {
+        requireFBO(fboConfig, staticFBM);
+    }
+
+    protected void requireDynamicFBO(FBOConfig fboConfig) {
+        requireFBO(fboConfig, dynamicFBM);
+    }
```

Comment:
I agree, but thinking from module developer perspective, I think it would be much more intuitive and easy. Rather then explicitly getting the FBM and using it as an argument. It was just for their convenience.